crates/bevy_state/src/lib.rs

```diff
@@ -34,6 +34,9 @@ pub mod app;
 pub mod condition;
 /// Provides definitions for the basic traits required by the state system
 pub mod state;
+#[cfg(feature = "bevy_hierarchy")]
+/// Provides [`StateScoped`] and [`clear_state_scoped_entities`] for managing lifetime of entities.
+pub mod state_scoped;
```

Comment:
why is this only with `bevy_hierarchy` enabled?

Reply:
Requires recursive despawns

Reply:
Requires recursive despawns.
It'll be weird if the state scope behaves differently based on a feature.
Do you think we should have 2 different methods?

Reply:
IMO we should despawn non-recursively without the feature. That's a good safe fallback.

Reply:
I would also prefer the non-recursive fallback